service/http/http.go

```diff
@@ -85,14 +100,24 @@ func New(opts Options) *Service {
 		n = opts.Clusterer.Node
 	}
 
+	var (
+		lazyNetLis = &lazyListener{}
+		lazyLis    = &lazyListener{}
+	)
+
+	// lazyLis should default to wrapping around lazyNetLis.
+	_ = lazyLis.SetInner(lazyNetLis)
```

Comment:
The `lazyLis` points to `lazyNetLis` which has `nil` `inner`. 

So I don't think that any of them is really usable, since the inner-most listener is `nil`. 

Since they are not usable until the `Update` is called, maybe we need only one `lazyListener` - the `lazyLis` and a simple reference to `Listener` that is nil in the beginning until `Run` is called where we would set it to the `netLis` from line 144 🤔 

In other words, I'm just wondering if we need two lazyListeners if one of them has `SetInner` called only once.

Reply:
> In other words, I'm just wondering if we need two lazyListeners if one of them has SetInner called only once.

The main reason is so we don't start listening for network traffic until the service runs. Doing otherwise feels like it breaks the conventions of a Service: don't do anything until Run is called.